darts/tests/models/forecasting/test_baseline_models.py

```diff
@@ -0,0 +1,418 @@
+import itertools
+
+import numpy as np
+import pytest
+
+from darts import TimeSeries
+from darts.logging import get_logger
+from darts.models import NaiveDrift, NaiveMean, NaiveMovingAverage, NaiveSeasonal
+from darts.models.forecasting.forecasting_model import (
+    GlobalForecastingModel,
+    LocalForecastingModel,
+)
+from darts.tests.conftest import tfm_kwargs
+from darts.utils import timeseries_generation as tg
+
+logger = get_logger(__name__)
+
+
+icl = 5
+local_models = [
+    (NaiveDrift, {}),
+    (NaiveMean, {}),
+    (NaiveMovingAverage, {}),
+    (NaiveSeasonal, {}),
+]
+global_models = []
+
+
+try:
+    import torch
+
+    from darts.models import GlobalNaiveAggregate, GlobalNaiveDrift, GlobalNaiveSeasonal
+
+    TORCH_AVAILABLE = True
+
+    global_models += [
+        (
+            GlobalNaiveAggregate,
+            {"input_chunk_length": icl, "output_chunk_length": 3, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveAggregate,
+            {"input_chunk_length": icl, "output_chunk_length": 1, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveDrift,
+            {"input_chunk_length": icl, "output_chunk_length": 3, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveDrift,
+            {"input_chunk_length": icl, "output_chunk_length": 1, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveSeasonal,
+            {"input_chunk_length": icl, "output_chunk_length": 3, **tfm_kwargs},
+        ),
+        (
+            GlobalNaiveSeasonal,
+            {"input_chunk_length": icl, "output_chunk_length": 1, **tfm_kwargs},
+        ),
+    ]
+
+    def custom_mean_valid(x, dim):
+        return torch.mean(x, dim)
+
+    def custom_mean_invalid_out_shape(x, dim):
+        return x[:1]
+
+    def custom_mean_invalid_signature(x):
+        return torch.mean(x, dim=1)
+
+    def custom_mean_invalid_output_type(x, dim):
+        return torch.mean(x, dim=1).detach().numpy()
+
+except ImportError:
+    logger.warning("Torch not installed - will be skipping Torch models tests")
+    TORCH_AVAILABLE = False
+
+    custom_mean_valid = None
+    custom_mean_invalid_out_shape = None
+    custom_mean_invalid_signature = None
+    custom_mean_invalid_output_type = None
+
+
+class TestBaselineModels:
```

Comment:
Missing tests for `output_chunk_shift` > 0

Reply:
It is tested in `test_torch_forecasting_model.test_output_shift`, and a dedicated test here for GlobalNaiveDrift (the only one for the shift will lead predict different results, e.g. Drift continues `output_chunk_shift` points)